acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/hashicorpVault/HashicorpVaultDocker.java

```diff
@@ -0,0 +1,250 @@
+
+  private final DockerClient docker;
+  private final String vaultContainerId;
+
+  private int port;
+  private String ipAddress;
+
+  public HashicorpVaultDocker(final DockerClient docker) {
+    this.docker = docker;
+    pullVaultImage();
+    this.vaultContainerId = createVaultContainer();
+  }
+
+  public void start() {
+    LOG.info("Starting Hashicorp Vault Docker container: {}", vaultContainerId);
+    docker.startContainerCmd(vaultContainerId).exec();
+
+    this.ipAddress = getDockerHostIp();
+    LOG.info("Docker Host IP address: {}", ipAddress);
+
+    LOG.info("Querying for the Docker dynamically allocated vault port number");
+    final InspectContainerResponse containerResponse =
+        docker.inspectContainerCmd(vaultContainerId).exec();
+
+    final Ports ports = containerResponse.getNetworkSettings().getPorts();
+    port = httpRpcPort(ports);
+    LOG.info("Http port for Hashicorp Vault: {}", port);
+  }
+
+  public void createTestData() {
+    LOG.info("creating the secret in vault that contains the private key.");
+    final ExecCreateCmdResponse execCreateCmdResponse =
+        getExecCreateCmdResponse(CREATE_ETHSIGNER_SIGNING_KEY_SECRET);
+    waitFor(
+        60,
+        () ->
+            assertThat(
+                    runCommandInVaultContainer(execCreateCmdResponse, EXPECTED_FOR_SECRET_CREATION))
+                .isTrue());
+    LOG.info("The secret was created successfully.");
+  }
+
+  private String getDockerHostIp() {
+    final DefaultDockerClientConfig dockerConfig =
+        DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+    final Optional<String> optional = Optional.of(dockerConfig.getDockerHost()).map(URI::getHost);
+    return optional.orElse(LOCALHOST);
+  }
+
+  private ExecCreateCmdResponse getExecCreateCmdResponse(final String[] commandWithArguments) {
+    return docker
+        .execCreateCmd(vaultContainerId)
+        .withAttachStdout(true)
+        .withAttachStderr(true)
+        .withCmd(commandWithArguments)
+        .exec();
+  }
+
+  private boolean runCommandInVaultContainer(
+      final ExecCreateCmdResponse execCreateCmdResponse, final String expectedInStdout)
+      throws InterruptedException {
+    final ByteArrayOutputStream stdout = new ByteArrayOutputStream();
+    final ByteArrayOutputStream stderr = new ByteArrayOutputStream();
+    final ExecStartResultCallback resultCallback = new ExecStartResultCallback(stdout, stderr);
+    LOG.info(
+        "execCreateCmdResponse with id: {}, containerId: {}",
+        execCreateCmdResponse.getId(),
+        vaultContainerId);
+    final ExecStartResultCallback execStartResultCallback =
+        docker.execStartCmd(execCreateCmdResponse.getId()).exec(resultCallback).awaitCompletion();
+    execStartResultCallback.onError(
+        new RuntimeException(
+            "command in Hashicorp Vault returned error\n" + execStartResultCallback.toString()));
+    if (expectedInStdout != null) {
```

Comment:
The two places this method is called there is always an `expectedInStdOut` input value (i.e. it is never `null`)  :. no need for the null check / collapse the code

Reply:
done